src/openrct2/PlatformEnvironment.cpp

```diff
@@ -21,20 +21,21 @@ using namespace OpenRCT2;
 class PlatformEnvironment final : public IPlatformEnvironment
 {
 private:
-    std::string _basePath[DIRBASE_COUNT];
+    std::string _underlyingBasePath[+DIRBASE::Count];
+    const EnumeratedArray<std::string, DIRBASE> _basePath{ _underlyingBasePath };
 
 public:
     explicit PlatformEnvironment(DIRBASE_VALUES basePaths)
     {
-        for (size_t i = 0; i < DIRBASE_COUNT; i++)
+        for (auto i = static_cast<DIRBASE>(0); i < DIRBASE::Count; i++)
```

Comment:
Would something like
```suggestion
        for (DIRBASE i = 0; i < DIRBASE::Count; i++)
```
work?

Reply:
`DIRBASE` is a strong enum so doesn't allow implicit conversion from an integer